File: swift/BCURUI/Sources/BCURUI/URRecognizedText.swift
import CoreGraphics

/// A block of text recognized by the camera's text recognition pipeline.
///
/// Bounding boxes use normalized coordinates (0-1) with a top-left origin,
/// matching UIKit's coordinate system.
public struct URRecognizedText: Sendable {
    /// The recognized text string.
    public let text: String

    /// Normalized bounding box (0-1, top-left origin) of the text in the camera frame.
    public let boundingBox: CGRect

    /// Recognition confidence (0-1).
    public let confidence: Float

    /// The rotation (in degrees, 0/90/180/270) at which the text was detected.
    public let rotation: Int

    public init(text: String, boundingBox: CGRect, confidence: Float, rotation: Int = 0) {
        self.text = text
        self.boundingBox = boundingBox
        self.confidence = confidence
        self.rotation = rotation
    }
}

File: kotlin/bc-urui/src/main/kotlin/com/blockchaincommons/bcurui/URRecognizedText.kt
package com.blockchaincommons.bcurui

import android.graphics.RectF

/**
 * A block of text recognized by the camera's text recognition pipeline.
 *
 * Bounding boxes use normalized coordinates (0-1) with a top-left origin.
 */
data class URRecognizedText(
    /** The recognized text string. */
    val text: String,

    /** Normalized bounding box (0-1, top-left origin) of the text in the camera frame. */
    val boundingBox: RectF,

    /** Recognition confidence (0-1). */
    val confidence: Float,

    /** The rotation (in degrees, 0/90/180/270) at which the text was detected. */
    val rotation: Int = 0
)
